localstack/aws/protocol/service_router.py

```diff
@@ -138,14 +138,19 @@ def custom_signing_name_rules(signing_name: str, path: str) -> Optional[ServiceM
 
 def custom_host_addressing_rules(host: str) -> Optional[ServiceModelIdentifier]:
```

Comment:
Probably a super edge case to ignore but I guess if someone uses a custom domain name containing one of these strings, we might route it to the wrong service 💥

Reply:
agree. this is actually a problem we have in lots of places where we do fuzzy url matching. :shrug: so far we haven't had anyone running into that issue though